src/auxiliary/math.jl

```diff
@@ -5,6 +5,93 @@
 @muladd begin
 #! format: noindent
 
+"""
+    sqrt_(x::Real)
+
+Custom square root function which returns `NaN` for negative arguments instead of throwing an error.
+This is required to ensure [correct results for multithreaded computations](https://github.com/trixi-framework/Trixi.jl/issues/1766) 
+when using the [`Polyester` package](https://github.com/JuliaSIMD/Polyester.jl), 
+i.e., using the @batch macro instead of the Julia built-in @threads macro, see [`@threaded`](@ref).
+
+We dispatch this function for `Float64, Float32, Float16` to the LLVM intrinsic `sqrt_llvm` as for this the 
+`sqrt_llvm` function can be used out-of the box, i.e., it returns `NaN` for negative arguments.
+For other types, such as integers or dual numbers required for algorithmic differentiation, we
+fall back to the Julia built-in `sqrt` function after a check for negative arguments.
+Since these cases are not performance critical, the check for negativity does not hurt here 
+and can (as of now) optimized away by the compiler.
+
+When debugging code, it might be useful to change the implementation of this function to redirect to 
+the Julia built-in `sqrt` function, as this reports the exact place in code where the domain is violated 
+in the stacktrace.
+"""
+@inline sqrt_(x::Real) = x < zero(x) ? oftype(x, NaN) : Base.sqrt(x)
+
+"""
+    sqrt_(x::Union{Float64, Float32, Float16})
+
+Custom square root function which returns `NaN` for negative arguments instead of throwing an error.
+This is required to ensure [correct results for multithreaded computations](https://github.com/trixi-framework/Trixi.jl/issues/1766) 
+when using the [`Polyester` package](https://github.com/JuliaSIMD/Polyester.jl), 
+i.e., using the @batch macro instead of the Julia built-in @threads macro, see [`@threaded`](@ref).
+
+For `Float64, Float32, Float16` the LLVM intrinsic `sqrt_llvm` function can be used out-of the box,
+i.e., it returns `NaN` for negative arguments.
+For other types, such as integers or dual numbers required for algorithmic differentiation, we
+fall back to the Julia built-in `sqrt` function after a check for negative arguments, 
+see [`sqrt_(x::Real)`](@ref).
+
+When debugging code, it might be useful to change the implementation of this function to redirect to 
+the Julia built-in `sqrt` function, as this reports the exact place in code where the domain is violated 
+in the stacktrace.
+"""
```

Comment:
```suggestion
```
Avoid having the same docstring twice?

Reply:
It is not exactly identical, but contains less information than the one for `x::Real`.

Reply:
But the information present is the same, isn't it? So it will be doubled if you use the REPL help mode or look at our online docs.